Google.Api.Gax.Grpc.Tests/ApiCallTest.cs

```diff
@@ -132,8 +136,310 @@ public void WithGoogleRequestParam()
             call1.Sync(new SimpleRequest { Name = "test" }, null);
             call1.Async(new SimpleRequest { Name = "test" }, null);
 
-            CallSettingsTest.AssertSingleHeader(syncCallSettings, CallSettings.RequestParamsHeader, "parent=test");
-            CallSettingsTest.AssertSingleHeader(asyncCallSettings, CallSettings.RequestParamsHeader, "parent=test");
+            var expectedHeader = "parent=test";
```

Comment:
I'd suggest doing the vast majority of these tests as unit tests of HeaderParameterExtractor - that keeps the test closer to the logic, and will reduce the setup required (and duplication of testing with sync and async). I would expect just a single test for each of ApiCall and ApiServerStreamingCall would be sufficient.

Reply:
My counter-argument would be that if we decide to move around parts when refactoring -- e.g. where does the lambda go, who is responsible for merging and escaping the parameters --  this file will give us a complete suite of tests that won't have to be modified, which is very helpful. That is why I would prefer to keep these here.

I'm not 100% married to the idea, so if you still think these should be moved I am open to it.

Reply:
I really think it'll be simpler to provide good coverage with focused tests. We should definitely keep *some* tests in here, as effectively integration tests - but they don't need to cover every detail.

I'm happy to make those test changes as a later refactoring PR though.

Reply:
👍